### PR TITLE
VRF V2 GA Documentation

### PR DESCRIPTION
**Pull request by dwightjl**
Preview on Vercel: https://documentation-git-fork-dwightjl-dwightjl-vrf-v2-chainlinklabs.vercel.app/docs/chainlink-vrf/

### CODE REVIEW

File: docs/Using Randomness/chainlink-vrf.md [hunk 1]
```diff
@@ -4,30 +4,108 @@ section: ethereum
...
+
+- **Subscription accounts:** An account that holds LINK tokens and makes them available to fund requests to Chainlink VRF v2 coordinators.
+- **Subscription owner:** The wallet address that creates and manages a subscription account. Any account can add LINK to the subscription balance, but only the owner can add approved consumers or withdraw funds.
+- **Consumers:** Contracts that are approved to use funding from your subscription account.
+- **Subscription balance:** The amount of LINK maintained on your subscription account. Requests from consumer contracts will continue to be funded until the balance runs out, so be sure to maintain sufficient funds in your subscription balance to pay for the requests and keep your applications running.
+
+The Subscription Manager system has the following limitations:
+
+- Each subscription supports up to 100 consumer contracts. If you need more than 100 consumers, create multiple subscriptions.
+- Each subscription must maintain a minimum balance in order to fund requests from consumer contracts. If your balance drops below this threshold
```
Comment (that1guy) [inline docs/Using Randomness/chainlink-vrf.md:71]: @dwightjl this is an incomplete sentence.

Comment (dwightjl) [reply docs/Using Randomness/chainlink-vrf.md:71]: Good catch. I moved the "request will go through automatically as long as the request was made in the last 24 hours" sentence up from the Subscription billing section to resolve this.

File: _includes/samples/VRF/VRFv2SubscriptionManager.sol [hunk 2]
```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: MIT
+// Example of a single consumer contract which owns the subscription.
+pragma solidity ^0.8.7;
+
+import "@chainlink/contracts/src/v0.8/interfaces/LinkTokenInterface.sol";
+import "@chainlink/contracts/src/v0.8/interfaces/VRFCoordinatorV2Interface.sol";
+import "@chainlink/contracts/src/v0.8/dev/VRFConsumerBaseV2.sol";
+
+contract VRFv2Consumer is VRFConsumerBaseV2 {
```
Comment (makramkd) [inline _includes/samples/VRF/VRFv2SubscriptionManager.sol:9]: The name of this contract is the same as the previous file - perhaps this should be VRFv2SubscriptionManager instead?

File: _includes/samples/VRF/VRFv2SubscriptionManager.sol [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  address link_token_contract = 0x01BE23585060835E02B77ef475b0Cc51aA1e0709;
+
+  // The gas lane to use, which specifies the maximum gas price to bump to.
+  // For a list of available gas lanes on each network,
+  // see https://docs.chain.link/docs/vrf-contracts/#configurations
+  bytes32 keyHash = 0xd89b2bf150e3b9e13446986e571fb9cab24b13cea0a43ea20a6049a85cc807cc;
+
+  // A reasonable default is 1000000, but this value could be different
+  // on other networks.
+  uint32 callbackGasLimit = 1000000;
```
Comment (makramkd) [inline _includes/samples/VRF/VRFv2SubscriptionManager.sol:27]: Ditto on max callback gas limit here, could drop to 100k instead.

File: docs/Using Randomness/chainlink-vrf.md [hunk 4]
```diff
@@ -4,30 +4,108 @@ section: ethereum
...
+
+- **Callback gas limit:** Specifies the maximum amount of gas you are willing to spend on the callback request. Define this limit by specifying the `callbackGasLimit` value in your request.
+
+Requests to Chainlink VRF v2 follow the [Request & Receive Data](/docs/request-and-receive-data/) cycle. The VRF coordinator processes the request and determines the final charge to your subscription using the following steps:
+
+1. You submit your request with a specified gas lane `keyHash` and the `callbackGasLimit`. If your request is urgent, specify a gas lane with a higher gas price limit. The `callbackGasLimit` depends on the size of your request. Generally, a limit of 100,000 gas is appropriate. Verification gas has a hard upper limit of 200,000 gas.
+
+1. The coordinator starts the transaction and bumps the gas price until the transaction is completed. The coordinator will not exceed the gas price limit of your selected gas lane. If your request cannot be completed at your specified gas limit, resubmit the request with a different gas lane or wait until gas prices fall below your current gas lane limits.
```
Comment (makramkd) [inline docs/Using Randomness/chainlink-vrf.md:93]: Is it intentional that all numbers in this numbered list are `1.`?

Comment (dwightjl) [reply docs/Using Randomness/chainlink-vrf.md:93]: Yes. Markdown auto-orders them. It's easier to insert and remove points if we don't manually manage the order.

File: _includes/samples/VRF/VRFv2Consumer.sol [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  address link = 0x01BE23585060835E02B77ef475b0Cc51aA1e0709;
+
+  // The gas lane to use, which specifies the maximum gas price to bump to.
+  // For a list of available gas lanes on each network,
+  // see https://docs.chain.link/docs/vrf-contracts/#configurations
+  bytes32 keyHash = 0xd89b2bf150e3b9e13446986e571fb9cab24b13cea0a43ea20a6049a85cc807cc;
+
+  // A reasonable default is 1000000, but this value could be different
+  // on other networks.
+  uint32 callbackGasLimit = 1000000;
```
[the remark this answers is not in the document]
Comment (dwightjl) [reply _includes/samples/VRF/VRFv2Consumer.sol]: I think I must have typo'd an extra zero in there. Good catch. Thank you! Fixed.

File: docs/Introduction/tutorials/advanced-tutorial.md [hunk 6]
```diff
@@ -15,6 +15,11 @@ metadata:
 >
 > This guide requires basic knowledge about smart contracts. If you are new to smart contract development, read the [Consuming Data Feeds](/docs/consuming-data-feeds/) and [Random Numbers](/docs/intermediates-tutorial/) guides before you begin.
 
+<!-- TODO: Update this guide for VRF v2 -->
```
Comment (ritika-b) [inline docs/Introduction/tutorials/advanced-tutorial.md:18]: Is this TODO still necessary?

File: docs/Using Randomness/example-contracts.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+> ℹ️ You are viewing the VRF v2 guide.
+>
+> If you are using v1, see the [VRF v1 guide](/docs/get-a-random-number/v1/).
+
+How you manage the subscription depends on your randomness needs. You can configure your subscriptions using the [Subscription Manager](https://vrf.chain.link) application, but these examples demonstrate how to create your subscription and register your applications programmatically. For these examples, the contract owns and manages the subscription. You can still view the subscriptions in the [Subscription Manager](https://vrf.chain.link) and any wallet can provide funding to those subscriptions.
+
+**Table of contents**
+
+- [Modifying subscriptions and configurations](#modifying-subscriptions-and-configurations)
```
Comment (ritika-b) [inline docs/Using Randomness/example-contracts.md:22]: Generally speaking, is this the casing we'll be using for subtitles? I've been titling them using MLA title casing so just want to confirm

**Review by ritika-b — APPROVED** [review]

File: docs/Using Randomness/example-contracts.md (hunk 7 above)
Comment (dwightjl) [reply docs/Using Randomness/example-contracts.md:22]: I'm not too particular about heading case, but you are right we should be consistent. This is the only subtitle style I've ever used in software documentation.

- https://cloud.google.com/compute/docs/instances/nested-virtualization/creating-nested-vms
- https://www.ibm.com/docs/en/SSAUZ9_12.1.0/adbugl14.pdf
-

File: docs/Introduction/tutorials/advanced-tutorial.md (hunk 6 above)
Comment (dwightjl) [reply docs/Introduction/tutorials/advanced-tutorial.md:18]: Removed

**Pull request merged**